nexus/src/app/instance.rs

```diff
@@ -1325,59 +1184,6 @@ impl super::Nexus {
         Ok(())
     }
 
-    /// Takes an updated instance state returned from a call to sled agent and
-    /// writes it back to the database.
-    ///
-    /// # Return value
-    ///
-    /// - `Ok((instance_updated, vmm_updated))` if no failures occurred. The
-    ///   tuple fields indicate which database records (if any) were updated.
-    ///   Note that it is possible for sled agent not to return an updated
-    ///   instance state from a particular API call. In that case, the `state`
-    ///   parameter is `None` and this routine returns `Ok((false, false))`.
-    /// - `Err` if an error occurred while writing state to the database. A
-    ///   database operation that succeeds but doesn't update anything (e.g.
-    ///   owing to an outdated generation number) will return `Ok`.
-    async fn write_returned_instance_state(
-        &self,
-        instance_id: &InstanceUuid,
-        state: Option<nexus::SledInstanceState>,
-    ) -> Result<InstanceUpdateResult, Error> {
-        slog::debug!(&self.log,
-                     "writing instance state returned from sled agent";
-                     "instance_id" => %instance_id,
-                     "new_state" => ?state);
-
-        if let Some(state) = state {
-            let update_result = self
-                .db_datastore
-                .instance_and_vmm_update_runtime(
-                    instance_id,
-                    &state.instance_state.into(),
-                    &state.propolis_id,
-                    &state.vmm_state.into(),
-                    &state.migration_state,
-                )
-                .await;
-
-            slog::debug!(&self.log,
-                         "attempted to write instance state from sled agent";
-                         "instance_id" => %instance_id,
-                         "propolis_id" => %state.propolis_id,
-                         "result" => ?update_result);
-
-            update_result
-        } else {
-            // There was no instance state to write back, so --- perhaps
-            // obviously --- nothing happened.
-            Ok(InstanceUpdateResult {
-                instance_updated: false,
-                vmm_updated: false,
-                migration_updated: None,
-            })
-        }
-    }
```

Comment:
The changes to `notify_instance_updated` in this PR ended up making it and `write_returned_instance_state` more or less identical: both would just write the VMM and migration states to CRDB and then potentially kick off an update saga. Therefore, `write_returned_instance_state` has gone to a happy farm upstate where it can run around all day with other `Nexus` methods.